corgi/core/models.py

```diff
@@ -923,19 +861,51 @@ def get_product_details(variant_names: tuple[str], stream_names: list[str]) -> d
 class ComponentQuerySet(models.QuerySet):
     """Helper methods to filter QuerySets of Components"""
 
-    def released_components(self, include: bool = True) -> models.QuerySet["Component"]:
+    def filter_latest_nevra_by_name(self, component_name: str) -> str:
+        nevras = self.filter(name=component_name).values_list("nevra", flat=True)
+        if nevras:
+            # Get the latest NVR using python. This only works for valid RPM NEVRAs
+            return sorted(nevras, key=functools.cmp_to_key(compare_packages))[-1]
+        else:
+            return ""
+
+    def latest_components(
+        self,
+        include: bool = True,
+    ) -> "ComponentQuerySet":
+        """Return components from latest builds."""
+        names = self.values_list("name", flat=True).order_by().distinct().iterator()
+        query = Q()
+        for name in names:
```

Comment:
Is this ever called on *all* components? I see in the manifest file we call it on the already filtered set of components of the stream or component we are generating the manifest for, but if this is exposed over the API to allow filtering on all components it will be very slow since it has to iterate over all component names, no?